### ISSUE TITLE
in_array assertions do not work with literals

### ISSUE THREAD

**Issue by Guuzen** [vimeo/psalm#6506]
https://psalm.dev/r/d25bc07558
In case of variable psalm shows RedundantCondition error but in case of string or constant there is no error.
Looks like Psalm\Internal\Analyzer\Statements\Expression\AssertionFinder::processFunctionCall do not handle case with literals at all, only with variables.

**Comment by orklah** [comment]
Yeah, the assertion system works by scraping and adding assertions to variable names. It's quite a backbone of the whole module, I think changing this would require lots of work and refactoring

**Comment by weirdan** [comment]
The snippet above now emits `TypeDoesNotContainType` in both cases (but a single `RedundantCondition`, for the reasons explained above).
